docs/clients/promtail/scraping.md

```diff
@@ -0,0 +1,211 @@
+
+* The `__path__` label is a special label which Promtail uses after discovery to
+  figure out where the file to read is located. Wildcards are allowed.
+
+* The label `filename` is added for every file found in `__path__` to ensure the
+  uniqueness of the streams. It is set to the absolute path of the file the line
+  was read from.
+
+### Kubernetes Discovery
+
+Note that while promtail can utilize the Kubernetes API to discover pods as
+targets, it can only use pods that are running on the same hostname of the
+machine that promtail is running on.
+
+This means that any time Kubernetes service discovery is used, there must be a
+`relabel_config` that creates the intermediate label `__host__` from
+`__meta_kubernetes_pod_node_name`:
+
+```yaml
+relabel_configs:
+  - source_labels: ['__meta_kubernetes_pod_node_name']
+    target_label: '__host__'
+```
+
+See [Relabeling](#relabeling) for more information.
+
+## Relabeling
+
+Each `scrape_configs` entry can contain a `relabel_configs` stanza.
+`relabel_configs` is a list of operations to transform the labels from discovery
+into another form.
+
+A single entry in `relabel_configs` can also reject targets by doing an `action:
+drop` if a label value matches a specified regex. When a target is dropped, the
+owning `scrape_config` will not process logs from that particular source.
+Other `scrape_configs` without the drop action reading from the same target
+may still use and forward logs from it to Loki.
+
+A common usecase of `relabel_configs` is to transform an internal label such
+as `__meta_kubernetes_*` into an intermediate internal label such as
+`__service__`. The intermediate internal label may then be dropped based on
+value or transformed to a final external label, such as `__job__`.
+
+### Examples
+
+* Drop the target if a label is empty:
+```yaml
+  - action: drop
+    regex: ^$
+    source_labels:
+    - __service__
+```
+* Drop the target if any of these labels contains a value:
+```yaml
+  - action: drop
+    regex: .+
+    separator: ''
+    source_labels:
+    - __meta_kubernetes_pod_label_name
+    - __meta_kubernetes_pod_label_app
+```
+* Rename an internal label into an external label so that it will be sent to Loki:
+```yaml
+  - action: replace
+    source_labels:
+    - __meta_kubernetes_namespace
+    target_label: namespace
+```
+* Convert all of the Kubernetes pod labels into external labels:
+```yaml
+  - action: labelmap
+    regex: __meta_kubernetes_pod_label_(.+)
+```
+
+Additional reading:
+
+ * [Julien Pivotto's slides from PromConf Munich, 2017](https://www.slideshare.net/roidelapluie/taking-advantage-of-prometheus-relabeling-109483749)
+
+## HTTP client options
+
+Promtail uses the Prometheus HTTP client implementation for all calls to Loki.
+Therefore it can be configured using the `client` stanza:
+
+```yaml
+client: [ <client_option> ]
+```
+
+Reference for `client_option`:
+
+```yaml
+# Sets the `url` of loki api push endpoint
```

Comment:
Can we repeat the same comment about using distributor's URL in microservices mode here?